packages/next-intl/src/middleware/middleware.tsx

```diff
@@ -68,15 +72,18 @@ export default function createMiddleware(config: MiddlewareConfig) {
     }
 
     function rewrite(url: string) {
-      return NextResponse.rewrite(new URL(url, request.url), getResponseInit());
+      return NextResponse.rewrite(
+        new URL(url, request.nextUrl.href),
+        getResponseInit()
+      );
```

Comment:
Is it necessary to rewrite to the external host here (same for the redirect)?

In your PR comment you mentioned:
> Consequently, the rewrite functionality is not behaving as expected

Can you explain what is not behaving as expected? My impression would be that your Next.js app already receives requests on an internal host, why can't it handle rewrites to the internal address too?

That being said, your point is true that the alternate links should point to the external host. I'm just wondering if we could limit the fix to choosing the correct host in `getAlternateLinksHeaderValue`, not having to modify anything outside of that file.

What do you think?

Reply:
I haven't read about what alternatelink does exactly, but I'm curious about why the domain needs to be included. Can't the desired functionality be achieved with just the pathname without including the domain?

Reply:
As far as I know, the domain needs to be included: https://developers.google.com/search/docs/specialty/international/localized-versions#http